src/OpalCompiler-Core/CompiledMethod.extension.st

```diff
@@ -36,14 +36,37 @@ CompiledMethod >> reformat [
 	self methodClass compile: self ast formattedCode classified: self protocol
 ]
 
+{ #category : '*OpalCompiler-Core' }
+CompiledMethod >> saveBcToASTCacheWithAST: aMethodAST [
+
+	| compiledBlockAST |
+	self propertyAt: #bcToASTCache ifPresent: [ :bcToASTCache | ^ self ].
+	self propertyAt: #bcToASTCache put: aMethodAST bcToASTCache.
```

Comment:
So only the first ast is cached?
What would happen if I put a first breakpoint (method v0 cached, v1 becomes current), then the current ast is cached, then I put a second breakpoint later (I need v1 to be cached and v2 becomes current), is the cache being updated or does it stay the same as v0?

Reply:
Installing a new breakpoint (a new metalink) creates each time  a new compiled method, so there is no problem. If the AST changes, a new compiled method is created. This is why I save it the `bcToASTCache` only once